libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -48,23 +63,30 @@ type addHostMetadata struct {
 	geoData mapstr.M
 	config  Config
 	logger  *logp.Logger
+	metrics metrics
 }
 
-const (
-	processorName = "add_host_metadata"
-)
-
 // New constructs a new add_host_metadata processor.
 func New(cfg *config.C) (processors.Processor, error) {
 	c := defaultConfig()
 	if err := cfg.Unpack(&c); err != nil {
 		return nil, fmt.Errorf("fail to unpack the %v configuration: %w", processorName, err)
 	}
 
+	// Logging and metrics (each processor instance has a unique ID).
+	var (
+		id     = int(instanceID.Inc())
```

Comment:
First time I've seen this way of getting around the namespace collision issue. CC @fearful-symmetry, this could help the version of this problem you have in the shipper.

Reply:
huh, interesting!